akd/agents/storm/storm.py

```diff
@@ -1,73 +1,116 @@
-from langgraph.checkpoint.memory import MemorySaver
+from functools import partial
+from typing import Optional
+
+from langchain_community.vectorstores import InMemoryVectorStore
+from langchain_huggingface import HuggingFaceEmbeddings
+from langchain_openai import ChatOpenAI
+from langgraph.checkpoint.memory import InMemorySaver
+from langgraph.checkpoint.serde.jsonplus import JsonPlusSerializer
 from langgraph.graph import END, START, StateGraph
-from langgraph.pregel import RetryPolicy
-from pydantic import Field, computed_field
+from langgraph.types import RetryPolicy
+from loguru import logger
+from pydantic import AliasChoices, AnyUrl, Field
 
 from akd._base import InputSchema, OutputSchema
 from akd.agents._base import BaseAgent, BaseAgentConfig
-from akd.configs.storm_config import STORM_SETTINGS, StormSettings
+from akd.agents.search.aspect_search import AspectSearchConfig
 
-from .config import initialise_storm_config
 from .nodes import (
     conduct_interviews,
-    hitl_editors,
     index_references,
     initialize_research,
     refine_outline,
     write_article,
     write_sections,
 )
-from .state import ResearchState
-from .utils.outline_utils import Perspectives
+from .structures import Outline, Perspectives, ResearchState
 
 
 class StormInputSchema(InputSchema):
     """Input schema for storm agent"""
 
     config: dict | None = Field(
-        None,
+        default={"configurable": {"thread_id": "default-thread"}},
```

Comment:
Use default factory otherwise this dictionary is a mutable dictionary, to remove any potential side effects